src/Uart/Uart.c

```diff
@@ -268,24 +268,29 @@ Uart_getRxFifoCount(Uart *const uart)
 	return count;
 }
 
-static inline void
-handleRxInterrupt(Uart *const uart)
+static inline bool
+handleRxInterrupt(Uart *const uart, int* const errCode)
 {
 	uint8_t data = (uint8_t)uart->reg->rhr;
 
 	if (uart->rxFifo == NULL) {
 		disableRxIrq(uart);
-		return;
+		return true;
+	}
+
+	if(!ByteFifo_push(uart->rxFifo, data)) {
+		returnError(errCode, Uart_ErrorCodes_Rx_Fifo_Full);
```

Comment:
Didn't we discuss returning at this point?

Reply:
Where is false returned?

Reply:
ACK